config/initializers/new_framework_defaults_6_0.rb

```diff
@@ -37,7 +37,7 @@
 # If you send mail in the background, job workers need to have a copy of
 # MailDeliveryJob to ensure all delivery jobs are processed properly.
 # Make sure your entire app is migrated and stable on 6.0 before using this setting.
-# Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
+Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
```

Comment:
Will we need special precautions to handle migration? e.g. first drain the deliveries and only then deploy new version?

Reply:
I've been checking this. We don't need to take any action because old Rails 5.2 mailer jobs can be performed by Rails 6. This is what I found:

- They added a breaking change to `DeliveryJob` (https://github.com/rails/rails/pull/34367/commits/60339da5bcb76489576321fa12e665f176d8d692), then reverted it (https://github.com/rails/rails/pull/34591/commits/f5050d998def98563f8fa4b381c09f563681f159). So Rails 5 mailer jobs can be performed by Rails 6, but with a deprecation warnnig, see (https://github.com/rails/rails/pull/34591)
- All actions that enqueued a `DeliveryJob` or `Parameterized::DeliveryJob` ini Rails 5, now enqueue a `MailDeliveryJob` in Rails 6.
  - For instance, sending a message as provider to all users.
- This is what I tried locally:
  1. Send an email as provider to all users from Rails 5. Sidekiq is not running.
     1. Rails 5 enqueues a `MessageWorker` job.
     2. Switch to Rails 6.
     3. Run Sidekiq -> The job completes successfully.
  2. From the Rails 5 console:
     1. Sidekiq is not running. Run:
     ```
     [1] pry(main)> mes = Message.last
     [2] pry(main)> PostOffice.message_notification(mes, mes.recipients.first).deliver_later
     ```
     2. Rails 5 enqueues an `ActionMailer::DeliveryJob`.
     3. Switch to Rails 6.
     4. Run Sidekiq -> The job completes successfully, but with a deprecation warning.